File: app/src/main/java/ru/yandex/subbota_job/notes/DraftStorage.java
package ru.yandex.subbota_job.notes;

import android.content.Context;
import android.text.TextUtils;

/**
 * Created by subbota on 27.09.2016.
 */

class DraftStorage {
    private Context mContext;
    private final String keyPath = this.getClass().toString() + "path";
    private final String keyContent = this.getClass().toString() + "content";
    private final String preferenceName = this.getClass().toString();
    public DraftStorage(Context context)
    {
        mContext = context;
    }
    public String getDraftPath()
    {
        return mContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE).getString(keyPath, "");
    }
    public String getDraftContent()
    {
        return mContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE).getString(keyContent, "");
    }
    public void saveDraft(String path, String content)
    {
        mContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE).edit()
                .putString(keyPath, path).putString(keyContent, content).apply();
    }
    public void clearDraft()
    {
        mContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE).edit().clear().apply();
    }
}

File: app/src/main/java/ru/yandex/subbota_job/notes/DraftStorage.kt
package ru.yandex.subbota_job.notes

import android.content.Context
import android.text.TextUtils

/**
 * Created by subbota on 27.09.2016.
 */

internal class DraftStorage(private val mContext: Context) {
	private val keyPath = this.javaClass.toString() + "path"
	private val keyContent = this.javaClass.toString() + "content"
	private val preferenceName = this.javaClass.toString()
	val draftPath: String
		get() = mContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE).getString(keyPath, "")
	val draftContent: String
		get() = mContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE).getString(keyContent, "")

	fun saveDraft(path: String, content: String) {
		mContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE).edit()
				.putString(keyPath, path).putString(keyContent, content).apply()
	}

	fun clearDraft() {
		mContext.getSharedPreferences(preferenceName, Context.MODE_PRIVATE).edit().clear().apply()
	}
}
